azurerm/internal/services/appplatform/spring_cloud_service_resource.go

```diff
@@ -69,6 +69,7 @@ func resourceArmSpringCloudService() *schema.Resource {
 			"network": {
 				Type:     schema.TypeList,
 				Optional: true,
+				Computed: true,
```

Comment:
why is this now Computed, this means it's not possible to remove it, which would be a breaking change?

Reply:
digging into the specifics for this, the larger question from #9346 is why only the `cidr_ranges` element is being set given all of the fields are set into the state: https://github.com/terraform-providers/terraform-provider-azurerm/blob/23539e6b36c719931db16f554c20ec85134d8226/azurerm/internal/services/appplatform/spring_cloud_service_resource.go#L801-L807

as such, rather than making this Computed I believe this may want the block omitting if all of these fields are empty, as this appears to be a bug in the API? We should also raise a bug in the API, since this is an API bug from what I can tell, unless there's other fields returned in this block by default?

Reply:
@tombuildsstuff 
first of all, this is a sample config which will cause the diff:
```
resource "azurerm_spring_cloud_service" "test" {
  name                = "sc-example"
  resource_group_name = azurerm_resource_group.test.name
  location            = azurerm_resource_group.test.location
}
```
Before the backend service supports "outbound_ip" block, in the response of read rest api, the `networkProfile` is empty, so in the flatten function, we check it's nil and return `[]interface{}{}`, everything is right.

Recently, "outbound_ip" block is returned in the rest api, the response is like the following:
```
{
  "properties": {
    ...
    "networkProfile": {
      "outboundIPs": {
        "publicIPs": [
          "20.49.108.33",
          "20.49.108.45"
        ]
      }
    }
  },
  ...
}
```
because the `networkProfile` is not nil, in the flatten func, we will end up return a struct like this:
```
[]interface{}{
		map[string]interface{}{
			"app_subnet_id":                          "",
			"service_runtime_subnet_id":              "",
			"cidr_ranges":                            [],
			"app_network_resource_group":             "",
			"service_runtime_network_resource_group": "",
		},
	}
```

It seems this is a bug in the terraform-plugin-sdk. If user does not specify the whole `network` block, it could regard the string "" as empty, but thinks [] is different

Reply:
there is an alternative way to fix this issue: before the return statement in the flatten function, we check the length of all fields. if all the length is 0, we directly return `[]interface{}{}`. In this way, we could avoid adding "computed: true"

Reply:
that is what we should do @njuCZ - detect the empty value and return nothing

Reply:
done